influxql/decoder.go

```diff
@@ -0,0 +1,188 @@
+package influxql
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/flux/array"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/tablebuilder"
+	"github.com/influxdata/flux/values"
+)
+
+// NewResultDecoder will construct a new result decoder for an influxql response.
+func NewResultDecoder(a *execute.Allocator) flux.MultiResultDecoder {
+	return &resultDecoder{a: a}
+}
+
+type resultDecoder struct {
+	a *execute.Allocator
+}
+
+func (dec *resultDecoder) Decode(r io.ReadCloser) (flux.ResultIterator, error) {
+	var resp Response
+	if err := json.NewDecoder(r).Decode(&resp); err != nil {
+		return nil, err
+	}
+	return &resultIterator{resp: &resp, a: dec.a}, nil
+}
+
+type resultIterator struct {
+	resp *Response
+	a    *execute.Allocator
+}
+
+func (ri *resultIterator) More() bool {
```

Comment:
The SourceDecoder interface is similar to what you've done here, but instead defines  `Connect/Fetch/Decode` functions.  We had considered implementing more of an iterator pattern like this but decided to keep it simple.  What do you think?